### PR TITLE
[ESSNTL-2739] Fixes schema validation bug

### PR DESCRIPTION
**Pull request by FabriciaDinizRH**
# Overview

This PR is being created to further address [ESSNTL-2739](https://issues.redhat.com/browse/ESSNTL-2739).
Fero found that when some specific canonical facts were missing, the kafka message was not produced due to the validation against `CanonicalFactsSchema` failing.

I didn't find a way to skip schema-level validations, so the data is not being validated against the schema but the returned value contains the equivalent fields from `_CANONICAL_FACTS_FIELDS`.

## PR Checklist

- [x] Keep PR title short, ideally under 72 characters
- [ ] Descriptive comments provided in complex code blocks
- [ ] Tests: validate optimal/expected output
- [ ] Tests: validate exceptions and failure scenarios
- [ ] Tests: edge cases
- [ ] Recovers or fails gracefully during potential resource outages (e.g. DB, Kafka)
- [ ] Uses [type hinting](https://docs.python.org/3/library/typing.html), if convenient
- [ ] Documentation, if this PR changes the way other services interact with host inventory
- [ ] Links to related PRs

## Secure Coding Practices Documentation Reference

You can find documentation on this checklist [here](https://github.com/RedHatInsights/secure-coding-checklist).

## Secure Coding Checklist

- [ ] Input Validation
- [ ] Output Encoding
- [ ] Authentication and Password Management
- [ ] Session Management
- [ ] Access Control
- [ ] Cryptographic Practices
- [ ] Error Handling and Logging
- [ ] Data Protection
- [ ] Communication Security
- [ ] System Configuration
- [ ] Database Security
- [ ] File Management
- [ ] Memory Management
- [ ] General Coding Practices

### CODE REVIEW

File: app/queue/queue.py [hunk 1]
```diff
@@ -136,6 +137,7 @@ def _build_minimal_host_info(host_data):
         "insights_id": host_data.get("insights_id"),
```
Comment (CodeHeeler) [inline app/queue/queue.py:137]: Shouldn't this insights_id be removed as you did in app/queue/notifications.py line 75 above as it is now within the canonical_facts?

**Review by CodeHeeler — CHANGES_REQUESTED** [review]

File: app/queue/queue.py (hunk 1 above)
Comment (FabriciaDinizRH) [reply app/queue/queue.py:137]: Yes! I have forgotten about this one, thanks

**Review by CodeHeeler — DISMISSED** [review]
Looks great, thanks!

**Review by CodeHeeler — DISMISSED** [review]
LGTM

**Comment by fstavela** [comment]
I'm still getting the same error:
```
[2022-09-20 09:03:34,904] [1] [140713125044480] [inventory.app.queue.queue] [ERROR] Validation error while adding or updating host: {'_schema': ['provider_type and provider_id are both required.']}
[2022-09-20 09:03:34,905] [1] [140713125044480] [inventory.app.queue.queue] [ERROR] Unable to process message
Traceback (most recent call last):
File "/opt/app-root/src/app/queue/queue.py", line 313, in event_loop
handler(message.value, event_producer, notification_event_producer=notification_event_producer)
File "<decorator-gen-6>", line 2, in handle_message
File "/opt/app-root/lib64/python3.8/site-packages/prometheus_client/context_managers.py", line 66, in wrapped
return func(*args, **kwargs)
File "/opt/app-root/src/app/queue/queue.py", line 298, in handle_message
send_kafka_error_message(notification_event_producer, host=host, detail=str(ve.detail))
File "/opt/app-root/src/app/queue/queue.py", line 333, in send_kafka_error_message
minimal_host = _build_minimal_host_info(host)
File "/opt/app-root/src/app/queue/queue.py", line 139, in _build_minimal_host_info
"canonical_facts": deserialize_canonical_facts(host_data, all=True),
File "/opt/app-root/src/app/serialization.py", line 62, in deserialize_canonical_facts
raise ValidationException(str(e.messages)) from None
app.exceptions.ValidationException: {'_schema': ['provider_type and provider_id are both required.']}
```
and the notification message is not produced

**Comment by FabriciaDiniz** [comment]
> I'm still getting the same error:
> 
> ```
> [2022-09-20 09:03:34,904] [1] [140713125044480] [inventory.app.queue.queue] [ERROR] Validation error while adding or updating host: {'_schema': ['provider_type and provider_id are both required.']}
> [2022-09-20 09:03:34,905] [1] [140713125044480] [inventory.app.queue.queue] [ERROR] Unable to process message
> Traceback (most recent call last):
> File "/opt/app-root/src/app/queue/queue.py", line 313, in event_loop
> handler(message.value, event_producer, notification_event_producer=notification_event_producer)
> File "<decorator-gen-6>", line 2, in handle_message
> File "/opt/app-root/lib64/python3.8/site-packages/prometheus_client/context_managers.py", line 66, in wrapped
> return func(*args, **kwargs)
> File "/opt/app-root/src/app/queue/queue.py", line 298, in handle_message
> send_kafka_error_message(notification_event_producer, host=host, detail=str(ve.detail))
> File "/opt/app-root/src/app/queue/queue.py", line 333, in send_kafka_error_message
> minimal_host = _build_minimal_host_info(host)
> File "/opt/app-root/src/app/queue/queue.py", line 139, in _build_minimal_host_info
> "canonical_facts": deserialize_canonical_facts(host_data, all=True),
> File "/opt/app-root/src/app/serialization.py", line 62, in deserialize_canonical_facts
> raise ValidationException(str(e.messages)) from None
> app.exceptions.ValidationException: {'_schema': ['provider_type and provider_id are both required.']}
> ```
> 
> and the notification message is not produced

weird, I was getting an error still, but the message was being produced ok. I'll check what's happening

**Review by kruai — DISMISSED** [review]

**Comment by fstavela** [comment]
I triggered the build pipeline again, because it looks like the image was not built correctly last time. I'm still not getting the notifications, but during the debugging I found out that the code in the latest image doesn't have the latest changes.

**Comment by fstavela** [comment]
@FabriciaDinizRH Messages are now produced and also `account_id` is present in the notifications.

One last thing is that `host_id` is still always `None`. I don't think this is a blocker, so I think we can merge this and release and fix the `host_id` problem in another PR.

**Review by fstavela — APPROVED** [review]

**Review by kruai — APPROVED** [review]

**Review by CodeHeeler — APPROVED** [review]
LGTM, thank you!

**Review by thearifismail — APPROVED** [review]
LGTM

**Pull request merged**